cognite/client/_api/user_profiles.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import annotations
+
+from typing import List, Sequence, cast
+
+from cognite.client._api_client import APIClient
+from cognite.client._constants import LIST_LIMIT_DEFAULT
+from cognite.client.data_classes.user_profiles import UserProfile, UserProfileList
+from cognite.client.utils._identifier import UserIdentifierSequence
+
+
+class UserProfilesAPI(APIClient):
+    _RESOURCE_PATH = "/profiles"
+
+    def me(self) -> UserProfile:
+        """`Retrieve your own user profile <https://developer.cognite.com/api#tag/User-profiles/paths/~1profiles~1me/get>`_
+
+        Retrieves the user profile of the principal issuing the request, i.e. the principal *this* CogniteClient was instantiated with.
+
+        Returns:
+            UserProfile: Your own user profile.
+
+        Raises:
+            CogniteAPIError: If this principal doesn't have a user profile, you get a not found (404) response code.
+
+        Examples:
+
+            Get your own user profile:
+
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.iam.user_profiles.me()
+        """
+        return UserProfile._load(self._get(self._RESOURCE_PATH + "/me").json())
+
+    def list(self, limit: int | None = LIST_LIMIT_DEFAULT, initial_cursor: str | None = None) -> UserProfileList:
```

Comment:
I do not think we should expose the cursor here. I think that is a detail that should be hidden by the SDK. This is what we have done for all other listing methods I am familiar with. 
```suggestion
    def list(self, limit: int | None = LIST_LIMIT_DEFAULT) -> UserProfileList:
```

Reply:
I was very in doubt on this one 😅 But when I saw that `_list` already had support for `initial_cursor` I thought "why not" 🤷‍♂️

Reply:
Removed `initial_cursor `